r/src/compute-exec.cpp

```diff
@@ -106,6 +108,15 @@ class ExecPlanReader : public arrow::RecordBatchReader {
       return arrow::Status::OK();
     }
 
+    // Check for cancellation and stop the plan if we have a request. When
+    // the ExecPlan supports passing a StopToken and handling this itself,
+    // this will be redundant.
+    ARROW_RETURN_NOT_OK(stop_token_.Poll());
```

Comment:
You do not need to call both `Poll()` and `IsStopRequested()`. The former will return an error if and only if the latter returns true.

Reply:
Which also means that, in its current form, this code will (almost) never call `StopProducing`.